### DISCUSSION TITLE
Adding additional fields to the dataset content type.

### DISCUSSION THREAD

**Discussion by owen-soak** [GetDKAN/dkan#4314] [General]
How would i go about allowing for additional fields to be included in the datasets? The UI doesn't provide any obvious indication on how to do this. 

Thanks!

**Comment by stefan-korn** [comment]
@owen-soak : It probably depends if you want to add additional fields/properties to the dataset metadata field or additonal traditional drupal fields to the data content type.

For additional fields/properties in the metadata you would edit the dataset schema: https://github.com/GetDKAN/dkan/blob/2.x/schema/collections/dataset.json

You can/should put the your own version of the schema to "drupalroot"/schema/collections (just copy over from the dkan module folder) and customize your schema there.

For traditional drupal fields you can add them the usual way: https://www.drupal.org/docs/administering-a-drupal-site/managing-content-0/working-with-content-types-and-fields

That said, adding the fields is one part of the game and that should work straightforward regarding the backend editing. But showing the fields in the frontend is probably more difficult, especially if you use the decoupled React frontend.

**Comment by janette** [comment]
@owen-soak 
Here are the docs pertaining to customizing the schema, let us know how we could improve it https://dkan.readthedocs.io/en/latest/user-guide/guide_custom_schemas.html
